x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlTranslatorHandler.java

```diff
@@ -124,4 +140,110 @@ static Query translate(InsensitiveEquals bc) {
             return new TermQuery(source, name, value.utf8ToString(), true);
         }
     }
+
+    public static class TrivialBinaryComparisons extends ExpressionTranslator<BinaryComparison> {
+        @Override
+        protected Query asQuery(BinaryComparison bc, TranslatorHandler handler) {
+            ExpressionTranslators.BinaryComparisons.checkBinaryComparison(bc);
+            Query translated = translate(bc, handler);
+            return translated == null ? null : handler.wrapFunctionQuery(bc, bc.left(), () -> translated);
```

Comment:
This leads to NPEs because we use the overridden `wrapFunctionQuery`. I could make it deal with nulls, but that could accidentally mask unintended null queries, no?

Reply:
I missed that it's using the current handler function. All good.